StreamChatUITestsAppUITests/Robots/UserRobot+Asserts.swift

```diff
@@ -367,6 +367,40 @@ extension UserRobot {
         XCTAssertEqual(author, actualAuthor, file: file, line: line)
         return self
     }
+    
+    @discardableResult
+    func assertScrollToBottomButton(
+        isVisible: Bool,
+        file: StaticString = #filePath,
+        line: UInt = #line
+    ) -> Self {
+        var btn = MessageListPage.scrollToBottomButton
+        btn = isVisible ? btn.wait() : btn.waitForDisappearance()
+        XCTAssertEqual(isVisible,
+                       btn.exists,
+                       "Scroll to bottom button should be \(isVisible ? "visible" : "hidden")",
+                       file: file,
+                       line: line)
+        return self
+    }
+    
+    @discardableResult
+    func assertScrollToBottomButtonUnreadCount(
+        _ expectedCount: Int,
+        file: StaticString = #filePath,
+        line: UInt = #line
+    ) -> Self {
+        let unreadCount = MessageListPage.scrollToBottomButtonUnreadCount
+        let unreadCountShouldBeVisible = expectedCount > 0
+        if unreadCountShouldBeVisible {
+            XCTAssertEqual("\(expectedCount)", unreadCount.wait().label)
+        } else {
+            if unreadCount.exists {
+                XCTAssertEqual("\(expectedCount)", unreadCount.label)
+            }
```

Comment:
Shouldn't you XCFail if unreadCount does not exist?

Reply:
```swift
if unreadCountShouldBeVisible {
    XCTAssertEqual("\(expectedCount)", unreadCount.wait().label)
} else {
    if unreadCount.exists {
        XCTAssertEqual("\(expectedCount)", unreadCount.label)
    }
}
```

Maybe I did not get your point here, but it actually will fail if there is no `unreadCount` when it should be there. 
`Else` statement is a bit hackish here because in some scenarios, we still keep `unreadCount` in the hierarchy even though it equals to `0`. I discussed it with @nuno-vieira, it's not a big deal, but we will probably fix this in the future.